src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -18,9 +18,9 @@ namespace Microsoft.CommonLanguageServerProtocol.Framework;
 public abstract class AbstractLanguageServer<TRequestContext> : ILifeCycleManager, IAsyncDisposable
 {
     private readonly JsonRpc _jsonRpc;
-    private IRequestExecutionQueue<TRequestContext>? _queue;
+    private readonly Lazy<IRequestExecutionQueue<TRequestContext>> _queue;
     protected readonly ILspLogger _logger;
-    private ILspServices? _lspServices;
+    private readonly Lazy<ILspServices> _lspServices;
```

Comment:
Non blocking: Just curious why we prefer Lazy here. If the handler is not invoked we would construct now during exit notification which could be process exit, vs exit/close, vs branch switch(?), unless of course the init and exit are loosely coupled. Wouldn't a singleton on the lspServices directly tied to process start be still preferable then?

Reply:
talked to @ryanbrandenburg - these are lazy because the implementations of this type may need to do custom initialization (in the virtual ConstructXXX calls).  We want to defer initialization of these types until the implementer has signaled tells us to create them.  I did consider putting them in ctor params, but it ends up messier due to limitations in ordering / using instance methods in this ctors.   

I added a comment on that.